Update smoothing.md (#17821)
DataDog/documentation: content/en/dashboards/functions/smoothing.md

## Weighted 
Note: Weighted() is only available when querying `SUM BY` on gauge type metrics. 

| Function       | Description                                                           | Example                        |
| :----          | :-------                                                              | :---------                     |
| `weighted()`   | Automatically removes noise when summing your metric while preserving the proper weight of transient tags. | `sum:(<GAUGE_METRIC_NAME>{*}).weighted()` |

The `weighted()` function accounts for the short-lived lifespan of transient, churning tag values when summing gauge metrics in space to prevent artificial spikes. 

This function is automatically appended to queries on gauges if both of the following conditions are met: 
1. The metric has a regular, consistent submission interval that is also specified on Metrics Summary
2. The metric is queried with `SUM by` (i.e. a query string that resembles `sum: mygaugemetric{*}`)

Here is an example graph of our original query with inaccurate spikes (in purple) and the query with the properly weighted calculation (in green): 

> {{< img src="dashboards/functions/smoothing/weighed.png" alt="autosmooth illustration" style="width:80%;">}}

Author: Oops! added
```suggestion
{{< img src="dashboards/functions/smoothing/weighed.jpg" alt="weighted illustration" style="width:80%;">}}
```

### How does .weighted() work? 
Every metrics query has a standard order of evaluation (quick review of that order [here][3]). For example, the following query is calculated as follows: 
`sum:kubernetes.cpu.requests{*} by {kube_container_name}.rollup(avg, 10)`

> 1. Time aggregation -- We first sum the values across all unique tag value combinations (aka timeseries) in time for each 10s rollup time interval. The number of unique tag value combinations is actually determined by the most volatile / high granularity tag, let's say `container_id`,  on this metric. 

Author: ```suggestion
1. Time aggregation -- We first sum the values for each timeseries (defined by a unique tag value combination) in time for each 10s rollup time interval. The number of unique tag value combinations is actually determined by the most volatile / high granularity tag, let's say `container_id`,  on this metric. 
```

2. Then per `kube_container_name` (space aggregation), we sum of all averaged values as a single representative value. So summing values for each `kube_container_name` is dependent upon the number of unique `container_id`s there are for each rollup interval.

> The `weighted()` function properly accounts for the lifespan of the churning `container_id` tag values when summing by `kube_container_name` for this gauge metric.

Reviewer: What does churning mean here? Are tag values overwritten or are new values being produced as a rapid rate? Is it indicating that these values are constantly changing?
```suggestion
The `weighted()` function accounts for the short lifespan of the `container_id` tag values when summing by `kube_container_name` for this gauge metric.
```

#### Example

> Let's suppose we have the following query and our gauge metric's submission interval is defined at 10s. 

Reviewer: ```suggestion
Consider this query with the following assumptions:
`sum:kubernetes.cpu.requests{*} by {shard}.rollup(avg, 60)`

- The gauge metric's submission interval is defined at 10 seconds. 
```

> `sum:kubernetes.cpu.requests{*} by {shard}.rollup(avg, 60)`

Reviewer: ```suggestion
```

> And we're graphing a datapoint every 60s in time. Also for simplicity, let's assume we have a kubernetes cluster with 2 pods at any given time. Each pod is labeled with a shard and there is only ever 1 pod per shard. So with a couple of deploys that switch out pods, our raw data over 60 seconds could resemble: 

Reviewer: ```suggestion
- A datapoint is graphed every 60 seconds in time.
- There is a Kubernetes cluster with 2 pods at any given time. Each pod is labeled with a shard and there is only ever 1 pod per shard. 

With a couple of deploys that switch out pods, the raw data over 60 seconds could resemble: 
```

(INSERT TABLE resembling: https://a.cl.ly/L1uDl1wy) 

1. _Time Aggregation -- Rolling up data_

> With time aggregation, we're rolling up data either `avg` (without weighted) or the proposed `weighted_avg`: 

Reviewer: ```suggestion
With time aggregation, data is rolled up with either `avg` (without weighted) or the proposed `weighted_avg`: 
```

Author: ```suggestion
With time aggregation, we're rolling up data either `avg` (without weighted) or the proposed `weighted` average: 
```

(INSERT TABLE HERE resembling: https://a.cl.ly/d5ugkK7P) 

2. _Space Aggregation_ 

> Finally, we aggregate by shard to get the final values below: 

Reviewer: ```suggestion
Finally, the metric is aggregated by shard to get the final values below: 
```

(INSERT TABLE HERE resembling: https://a.cl.ly/YEueYxq1)

> Here we can see `weighted()` provides a weighted average of each of the two shards according to the amount of time they were submitting data. 

Reviewer: ```suggestion
The `weighted()` modifier provides a weighted average of each of the two shards according to the amount of time they were submitting data. 
```
